src/stores/PreferenceStore.ts

```diff
@@ -1,42 +1,73 @@
-import {action, observable} from "mobx";
-import {FrameScaling, RenderConfigStore} from "stores/RenderConfigStore";
+import {action} from "mobx";
+import * as AST from "ast_wrapper";
+import {FrameScaling, RenderConfigStore} from "stores";
 
-export class PreferenceStore {
-    // default settings json
-    private static readonly DEFAULT_SETTINGS: string = `{
-        "scaling": 0
-    }`;
-    @observable json;
+const PREFERENCE_KEYS = {
+    scaling: "CARTA_scaling",
+    colormap: "CARTA_colormap",
+    percentile: "CARTA_percentile",
+    astColor: "CARTA_astColor"
+};
+
+const DEFAULTS = {
+    scaling: 0,
+    colormap: "inferno",
+    percentile: 99.9,
+    astColor: 4
+};
 
+export class PreferenceStore {
     // user configurable settings
+    validateScaling(scaling: string) {
+        const value = Number(scaling);
+        return scaling && value !== NaN && RenderConfigStore.SCALING_TYPES.has(value) ? value : null;
```

Comment:
`value !== NaN` is always true, even for `NaN`, since `NaN === NaN` is false. Rather use `isfinite(value)`